includes/multi-currency/RestController.php

```diff
@@ -89,95 +140,79 @@ public function register_routes() {
 	}
 
 	/**
 	 * Gets the currency settings for a single currency.
 	 *
-	 * @param   \WP_REST_Request $request  Full data about the request.
+	 * @param \WP_REST_Request $request Full data about the request.
 	 *
-	 * @return  array            The currency settings.
+	 * @return \WP_REST_Response|\WP_Error The single currency settings as an array.
 	 */
-	public function get_currency_settings( $request ) {
-		$currency_code = sanitize_key( strtolower( $request['currency_code'] ) );
-		return [
-			'exchange_rate_type' => get_option( 'wcpay_multi_currency_exchange_rate_' . $currency_code, 'automatic' ),
-			'manual_rate'        => get_option( 'wcpay_multi_currency_manual_rate_' . $currency_code, null ),
-			'price_rounding'     => get_option( 'wcpay_multi_currency_price_rounding_' . $currency_code, null ),
-			'price_charm'        => get_option( 'wcpay_multi_currency_price_charm_' . $currency_code, null ),
-		];
+	public function get_single_currency_settings( $request ) {
+		$currency_code = $request->get_param( 'currency_code' );
+		return rest_ensure_response( WC_Payments_Multi_Currency()->get_single_currency_settings( $currency_code ) );
 	}
 
 	/**
 	 * Updates the currency settings for a single currency.
 	 *
-	 * @param   \WP_REST_Request $request  Full data about the request.
+	 * @param \WP_REST_Request $request Full data about the request.
 	 *
-	 * @return  array            The currency settings.
+	 * @return \WP_REST_Response|\WP_Error The single currency settings as an array.
 	 */
-	public function update_currency_settings( $request ) {
-		$currency_code        = sanitize_key( strtolower( $request['currency_code'] ) );
-		$params               = $request->get_params();
-		$available_currencies = WC_Payments_Multi_Currency()->get_available_currencies();
-
-		if ( array_key_exists( strtoupper( $currency_code ), $available_currencies ) ) {
-			if ( isset( $params['exchange_rate_type'] ) && in_array( $params['exchange_rate_type'], [ 'automatic', 'manual' ], true ) ) {
-				update_option( 'wcpay_multi_currency_exchange_rate_' . $currency_code, esc_attr( $params['exchange_rate_type'] ) );
-			}
-			if ( 'manual' === $params['exchange_rate_type'] && isset( $params['manual_rate'] ) ) {
-				update_option( 'wcpay_multi_currency_manual_rate_' . $currency_code, (float) $params['manual_rate'] );
-			}
-			if ( isset( $params['price_rounding'] ) ) {
-				update_option( 'wcpay_multi_currency_price_rounding_' . $currency_code, (float) $params['price_rounding'] );
-			}
-			if ( isset( $params['price_charm'] ) ) {
-				update_option( 'wcpay_multi_currency_price_charm_' . $currency_code, (float) $params['price_charm'] );
-			}
-		}
-
-		return $this->get_currency_settings( $request );
+	public function update_single_currency_settings( $request ) {
+		$currency_code      = $request->get_param( 'currency_code' );
+		$exchange_rate_type = $request->get_param( 'exchange_rate_type' );
+		$price_rounding     = $request->get_param( 'price_rounding' );
```

Comment:
`price_rounding` and `price_charm` are expected as `float`s in the method signature of `update_single_currency_settings`, maybe we should ensure that they are converted to a float?

Reply:
This happens within the REST request itself with the addition here: https://github.com/Automattic/woocommerce-payments/pull/6468/files#diff-f5ba94e7313766ad1445e023461ba80603c3abbbe1f8ef0145115dccdda54472R99-R109

The conversion/casting happens here: https://github.com/WordPress/WordPress/blob/6.2.2/wp-includes/rest-api.php#L2795-L2797

We can add the casting back that I removed if you think it is needed.

Reply:
Gotcha! Thanks for the context, missed that it's done on the REST validation level.